src/lib.rs

```diff
@@ -120,6 +122,7 @@ impl fmt::Display for Error {
             Error::CannotRecoverAdaptorSecret => "failed to recover adaptor secret",
             Error::CannotVerifyAdaptorSignature => "failed to verify adaptor signature",
             Error::Upstream(inner) => return write!(f, "{}", inner),
+            Error::InvalidTweak => "Tweak must of size 32",
```

Comment:
I am a bit worried that being more specific in the `Display` impl compared to the variant name is going to cause headaches down the line where someone reuses `Error::InvalidTweak` without reading what it will print as.

Can we rename this variant to be `InvalidTweakLength`? I think that might also make handling this further down the line easier because that is often something that can be guaranteed and matching / unwrapping a specific error is easier compared to something as generic as `InvalidTweak`.